### PR TITLE
Use pre-commit.

### PR DESCRIPTION
**Pull request by mauch**
This adds pre-commit to the repo. I have followed the same method as used in aerial-conversion.

### CODE REVIEW

File: setup.cfg
Comment (sahandv) [inline setup.cfg:1]: If I recall correctly, there was a situation in which we could get errors and one of those infinite loops between repositories. 
I don't recall whether it was black or doc formatter, but it happened. So we may want to go back to `ignore = E501, W503, E203, E265` just to be on the safe side.

**Review by sahandv — CHANGES_REQUESTED** [review]
Everything seems fine and logical. My only concern is with the flake8 ignores, as commented. for now, we may use the original one to be on the safe side if you also agree.

File: setup.cfg
Comment (mauch) [reply setup.cfg:1]: OK. I've added E265 back in to be on the safe side.

**Review by sahandv — APPROVED** [review]
Seems fine. Thanks!

**Pull request merged**